### PR TITLE
Separate faf-db [MySQL] from faf-db-migrations [Flyway]

### PR DESCRIPTION
**Pull request by Brutus5000**
We still need to update the java-api and remove the version number before go live.
Also the image of faf-db-migrations needs to be named properly.

However, this state is ready for review.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,12 @@
+sudo: required
 dist: trusty
 
 services:
   - docker
 
 # Temporarily update docker until travis supports compose v3.3+ files
 before_install:
-  - sudo service mysql stop
+  - sudo /etc/init.d/mysql stop
```
Comment (micheljung) [inline .travis.yml]: Why?

File: .travis.yml [hunk 2]
```diff
@@ -21,15 +22,25 @@ install:
...
   - docker-compose up -d faf-java-api
-  - sleep 60
+  - |
+    echo -n "Waiting for faf-java-api "
+    while ! curl -s --max-time 1 http://localhost:8010 &> /dev/null
+    do
+      echo -n "."
+      sleep 1
+    done
   - docker-compose up -d faf-website
```
Comment (micheljung) [inline .travis.yml]: You could start `faf-website` at the same time as `faf-java-api` but idk if that would speed or slow down the startup process.

File: docker-compose.yml [hunk 3]
```diff
@@ -44,9 +44,31 @@ services:
...
     depends_on:
       - faf-init-volumes
+    healthcheck:
+      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
+      timeout: 20s
+      retries: 10
+
+  #
+  # FAF MySQL database migrations
+  # (this creates & updates the database tables and some required data)
```
Comment (micheljung) [inline docker-compose.yml]: None of the other services puts the explanation on a newline into parentheses

File: docker-compose.yml [hunk 4]
```diff
@@ -44,9 +44,31 @@ services:
...
+  #
+  faf-db-migrations:
+    container_name: faf-db-migrations
+    image: faforever/faf-db-migrations:v59.1-flyway
+    user: ${FAF_DB_USER}
+    networks:
+      faf:
+        aliases:
+        - "faf-db-migrations"
```
Comment (micheljung) [inline docker-compose.yml]: A network alias is only necessary if the service needs to be accessed by other services via docker network

**Review by micheljung — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (Brutus5000) [reply .travis.yml]: It was broken at some point. Will check if it works again.

**Pull request merged**